test/e2e-go/features/transactions/accountv2_test.go

```diff
@@ -285,16 +287,23 @@ int 1
 	a.NoError(err)
 	signedTxn, err = client.SignTransactionWithWallet(wh, nil, tx)
 	a.NoError(err)
-	_, err = client.BroadcastTransaction(signedTxn)
-	a.NoError(err)
-	round, err = client.CurrentRound()
-	a.NoError(err)
-	_, err = client.WaitForRound(round + 2)
-	a.NoError(err)
-	// Ensure the txn committed
-	resp, err = client.GetPendingTransactions(2)
+	txid, err = client.BroadcastTransaction(signedTxn)
 	a.NoError(err)
-	a.Equal(uint64(0), resp.TotalTxns)
+	for {
+		round, err = client.CurrentRound()
+		a.NoError(err)
+		_, err = client.WaitForRound(round + 1)
+		a.NoError(err)
+		// Ensure the txn committed
+		resp, err = client.GetPendingTransactions(2)
+		a.NoError(err)
+		if resp.TotalTxns == 1 {
+			a.Equal(resp.TruncatedTxns.Transactions[0].TxID, txid)
+			continue
+		}
+		a.Equal(uint64(0), resp.TotalTxns)
+		break
+	}
```

Comment:
checkEvalDelta on line 318 will still impose certain expectations on the round / test timing.